### PR TITLE
Enable foldable sidebar menu

### PR DESCRIPTION
**Pull request by FOWind**
Related issues: #31817 

Set `sidbebar_menu_foldabe = true` for Docsy `[params.ui]`

Before:

image

After:

image

### CODE REVIEW

**Comment by sftim** [comment]
I wonder: would it be better to introduce the new UI with the v1.24 release?

File: config.toml [hunk 1]
```diff
@@ -217,6 +217,7 @@ url = "https://v1-19.docs.kubernetes.io"
 [params.ui]
 # Enable to show the side bar menu in its compact state.
 sidebar_menu_compact = false
+sidebar_menu_foldable = true
```
Comment (championshuttler) [inline config.toml:220]: Can we add a comment as well what it does like the other keys

Comment (FOWind) [reply config.toml:221]: Have added

**Comment by tengqm** [comment]
I agree to @sftim that maybe we want to target this at 1.24?

File: config.toml [hunk 2]
```diff
@@ -217,6 +217,8 @@ url = "https://v1-19.docs.kubernetes.io"
 [params.ui]
 # Enable to show the side bar menu in its compact state.
 sidebar_menu_compact = false
+# Show expand/collapse icon for sidebar sections
```
Comment (championshuttler) [inline config.toml:220]:
```suggestion
# Show expand/collapse icon for sidebar sections.
```

Comment (championshuttler) [reply config.toml:220]: Minor nit

**Comment by kbhawkey** [comment]
A preview page:
https://deploy-preview-31818--kubernetes-io-main-staging.netlify.app/docs/setup/production-environment/tools/kops/

This navigation style looks like the original navigation tree.

**Comment by tengqm** [comment]
@FOWind Please retarget this at the dev-1.24 branch?

**Comment by FOWind** [comment]
> @FOWind Please retarget this at the dev-1.24 branch?

Done

**Comment by sftim** [comment]
Let's ship it.
Thanks

/retitle Enable foldable sidebar meny

/label tide/merge-method-squash
/lgtm

**Pull request merged**